frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.unit.spec.tsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import userEvent from "@testing-library/user-event";
+import { render, screen } from "__support__/ui";
+import { createEntitiesState } from "__support__/store";
+import { getMetadata } from "metabase/selectors/metadata";
+import { createMockState } from "metabase-types/store/mocks";
+import {
+  createSampleDatabase,
+  ORDERS_ID,
+} from "metabase-types/api/mocks/presets";
+import { Expression } from "metabase-types/types/Query";
+import ExpressionStep, { ExpressionStepProps } from "./ExpressionStep";
+
+describe("Notebook Editor > Expression Step", () => {
+  it("should handle updating existing expression", async () => {
+    const expression: Expression = ["abs", ["field", 17, null]];
+    const {
+      props: { updateQuery },
+      mocks: { addExpression, updateExpression },
+    } = setup(undefined, {
+      // add an existing custom column expression
+      "old name": expression,
+    });
+
+    userEvent.click(screen.getByText("old name"));
+
+    const nameField = await screen.findByPlaceholderText(
+      "Something nice and descriptive",
+    );
+
+    userEvent.type(nameField, "{selectall}{backspace}new name{enter}");
```

Comment:
What do you think about using [`clear`](https://testing-library.com/docs/ecosystem-user-event/#clearelement) to make this more concise?

```suggestion
    userEvent.clear(nameField);
    userEvent.type(nameField, "new name{enter}");
```